scripts/profiler/events.py

```diff
@@ -47,6 +46,37 @@ def __init__(self, submit, start_time, end_time):
         self.proc_start_time = start_time
         self.proc_end_time = end_time
 
+    def serialize(self):
+        si = StringIO()
+        sw = csv.writer(si, delimiter=',')
+        sw.writerow(list((self.submit.type_id,
+                    self.submit.timestamp,
+                    self.proc_start_time,
+                    self.proc_end_time,
+                    *self.submit.data)))
+        return si.getvalue().strip('\r\n')
+
+    @staticmethod
+    def deserialize(scsv):
+        f = StringIO(scsv)
+        reader = csv.reader(f, delimiter=',')
+        #There should be only one row
```

Comment:
You could assert this and remove for loop then

Reply:
`reader` is an iterator object so getting its length is not that easy.